MSAL/src/native_auth/public/MSALNativeAuthPublicClientApplication.swift

```diff
@@ -30,6 +30,9 @@ public final class MSALNativeAuthPublicClientApplication: MSALPublicClientApplic
     let controllerFactory: MSALNativeAuthControllerBuildable
     let inputValidator: MSALNativeAuthInputValidating
     private let internalChallengeTypes: [MSALNativeAuthInternalChallengeType]
+    
+    lazy var cacheAccessor =
```

Comment:
If we want that only one instance of `MSALNativeAuthCacheAccessor` is created and used, why we don't use the Singleton pattern for this class? This can simplify the injection of the dependency

Reply:
There could be multiple instances of the cache accessor at runtime if there are multiple instances of the `MSALNativeAuthPublicClientApplication` class and that is fine, so I don't see the use for a singleton here.
The reason we use lazy initialisation (unlike how `inputValidator` is initialised for example) is because the `tokenCache` and `accountMetadataCache` members are not valid until the `MSALPublicClientApplication` initialiser has been called (i.e. `super.init(...)`) so we can't do it in the regular initialisers. I'm happy to discuss more if that doesn't answer your question or you have an alternative.

Reply:
I'm wondering if makes sense to have more than one instance of this class, even if we have multiple instances of the `MSALNativeAuthPublicClientApplication` class. Given that multiple instantiations of this class created the issue that we're trying to solve in this PR, I'm asking if it is ok to have multiple instances of the cacheAccessor.

Reply:
After a chat we agreed that the singleton pattern here is not ideal. We're using the same approach of existing MSAL, that is: one cache accessor per public client application instance